cli/operator/node.go

```diff
@@ -60,9 +60,16 @@ type KeyStore struct {
 	PasswordFile   string `yaml:"PasswordFile" env:"PASSWORD_FILE" env-description:"Password for operator private key file decryption"`
 }
 
+type DBOptions struct {
+	Path       string        `yaml:"Path" env:"DB_PATH" env-default:"./data/db" env-description:"Path for storage"`
+	Reporting  bool          `yaml:"Reporting" env:"DB_REPORTING" env-default:"false" env-description:"Flag to run on-off db size reporting"`
+	GCInterval time.Duration `yaml:"GCInterval" env:"DB_GC_INTERVAL" env-default:"6m" env-description:"Interval between garbage collection cycles. Set to 0 to disable."`
+}
+
 type config struct {
 	global_config.GlobalConfig `yaml:"global"`
-	DBOptions                  basedb.Options                   `yaml:"db"`
+	DBOptions                  DBOptions                        `yaml:"db"`
```

Comment:
can't we just use `basedb.Options`?

Reply:
The introduction of the new `DBOptions` struct serves to segregate the configurations for the node database and the slashing protection database. This separation is essential as `basedb.Options` could be employed for either database. By having a distinct `DBOptions` struct, we can maintain clearer and more organized configurations in our yaml/environment settings specifically for the slashing protection and node databases.

Reply:
honestly I don't understand your point. Since the struct is similar, I don't see how is this better than what we had before, maybe I'm missing something. can you explain further ?

Reply:
Our node should handle two different databases - one for the node and one for slashing protection. Each of these databases has its own set of configurations. 

The `basedb.Options` struct could technically be used for both, but to keep configurations separate and self-explanatory, I introduced two new structs - `DBOptions` for the node database and `SlashingProtectionOptions` for the slashing protection database.

Here’s an example with the `SlashingProtectionOptions` struct:
```go
type SlashingProtectionOptions struct {
    DBPath string yaml:"DBPath" env:"SP_DB_PATH" env-description:"Path for slashing protection db"
}
```
This struct allows us to define a specific path for the slashing protection database through yaml or environment variables, making it explicit what this configuration is for. 

Similarly, `DBOptions` is intended for configuring the node database. By having separate structs, we can avoid any potential confusion or misconfiguration that might arise from using a single struct for both databases. It also provides a clearer, more organized way to handle configurations for each database independently, making the code more readable and maintainable.

Reply:
I'm ok with this approach but IMO this isn't a lot better than just having the same struct type used twice for different purposes.